### PR TITLE
End of support for LZMA compression in writer

### LINKED ISSUE #714: Stop allowing to create ZIM file using LZMA
ZSTD works fine and is better than LZMA. Ultimatively we should remove support of LZMA to simplify the build/code and because there is fundamentally no reason to keep. The longer we or someone creates ZIM files using LZMA, the longer we will have to postpone the decision to fully remove the support LZMA in reading. Therefore we should IMO remove the LZMA support in writing (so in the creator) ASAP.

### PR DESCRIPTION
**Pull request by veloman-yunkan**
Fixes #714

### CODE REVIEW

**Comment by kelson42** [comment]
@veloman-yunkan Why removing code leads to code coverage diminushing? Looks wrong to me!

**Comment by veloman-yunkan** [comment]
> @veloman-yunkan Why removing code leads to code coverage diminushing? Looks wrong to me!

I see two reasons for that

**Comment by kelson42** [comment]
@veloman-yunkan We test properly the compression with zstd?

**Comment by veloman-yunkan** [comment]
> @veloman-yunkan We test properly the compression with zstd?

@kelson42 The answer depends on the implied meaning of the word "properly". I would say that some minimum expectations from testing are met.

**Review by mgautierfr — CHANGES_REQUESTED** [review]
We must also adapt the public API. At least:
- The compression enum in zim.h
- Remove the warning in `zim::writer::Creator::configCompression`

**Comment by veloman-yunkan** [comment]
> We must also adapt the public API. At least:
> 
>     * The compression enum in zim.h
> 
>     * Remove the warning in `zim::writer::Creator::configCompression`

LZMA-compressed ZIMs are still supported. Shouldn't we keep `Compression::Lzma` until we fully discontinue support for LZMA? Removing it now requires a hack in `getClusterReader()`.
 
I made the requested change in two steps so that the second one can be undone if we decide to keep `Compression::Lzma` (without supporting it during ZIM creation) a bit longer.

**Comment by mgautierfr** [comment]
The `zim::Compression` enum is part of the public API, it is used to configure the creator and we never use it publicly in the reader part (we don't tell how the zim archive has been compressed). So we should remove `Compression::Lzma`.

However, it is indeed a bit ugly to replace enum value with magic value and comment. It may be better to introduce a private "reading part" enum with all the compression we support (and have supported) and use this enum in `src/cluster.cpp`.

**Comment by veloman-yunkan** [comment]
> However, it is indeed a bit ugly to replace enum value with magic value and comment. It may be better to introduce a private "reading part" enum with all the compression we support (and have supported) and use this enum in `src/cluster.cpp`.

Done

File: src/cluster.cpp [hunk 1]
```diff
@@ -47,29 +47,29 @@ namespace
...
   uint8_t clusterInfo = zimReader.read(offset);
   // Very old zim files used 0 as a "default" compression, which means no compression.
   uint8_t compInfo = clusterInfo & 0x0F;
   if(compInfo == 0) {
-    *comp = Compression::None;
+    *comp = Cluster::Compression::None;
   } else if (compInfo == 2 /* Zip compression */) {
     throw std::runtime_error("zlib not enabled in this library");
   } else if (compInfo == 3 /* Bzip2 compression */) {
     throw std::runtime_error("bzip2 not enabled in this library");
```
Comment (mgautierfr) [inline src/cluster.cpp:60]: We should use `Cluster::Compression` enum here.

Comment (veloman-yunkan) [reply src/cluster.cpp]: Fixed

**Comment by mgautierfr** [comment]
Comparaison of compression have to be fixed (see the CI) but else I agree with the code.
You can rebase-fixup.

**Comment by veloman-yunkan** [comment]
Done

**Review by mgautierfr — APPROVED** [review]

**Pull request merged**